docs/contributor/image_mode.md

```diff
@@ -0,0 +1,369 @@
+* Store a `bootc` image in a remote registry and use it for installing a new RHEL
+operating system
+
+> See [Image mode for Red Hat Enterprise Linux](https://developers.redhat.com/products/rhel-image-mode/overview)
+for more information.
+
+The procedures described below require the following setup:
+* A `RHEL 9.4 host` with an active Red Hat subscription for building MicroShift `bootc`
+images and running containers
+* A `physical hypervisor host` with the [libvirt](https://libvirt.org/) virtualization
+platform for starting virtual machines that run RHEL OS containing MicroShift binaries
+* A `remote registry` (e.g. `quay.io`) for storing and accessing `bootc` images
+
+## Build MicroShift Bootc Image
+
+Log into the `RHEL 9.4 host` using the user credentials that have SUDO
+permissions configured.
+
+### Build Image
+
+Create the `Containerfile` file with the following contents.
+
+```
+FROM registry.redhat.io/rhel9/rhel-bootc:9.4
+
+ARG USHIFT_VER=4.16
+RUN dnf config-manager \
+        --set-enabled rhocp-${USHIFT_VER}-for-rhel-9-$(uname -m)-rpms \
+        --set-enabled fast-datapath-for-rhel-9-$(uname -m)-rpms
+RUN dnf install -y firewalld microshift && \
+    systemctl enable microshift && \
+    dnf clean all
+
+# Create a default 'redhat' user with the specified password.
+# Add it to the 'wheel' group to allow for running sudo commands.
+ARG USER_PASSWD
+RUN if [ -z "${USER_PASSWD}" ] ; then \
+        echo USER_PASSWD is a mandatory build argument && exit 1 ; \
+    fi
+RUN useradd -m -d /var/home/redhat -G wheel redhat && \
+    echo "redhat:${USER_PASSWD}" | chpasswd
+
+# Mandatory firewall configuration
+RUN firewall-offline-cmd --zone=public --add-port=22/tcp && \
+    firewall-offline-cmd --zone=trusted --add-source=10.42.0.0/16 && \
+    firewall-offline-cmd --zone=trusted --add-source=169.254.169.1
+# Application-specific firewall configuration
+RUN firewall-offline-cmd --zone=public --add-port=80/tcp && \
+    firewall-offline-cmd --zone=public --add-port=443/tcp && \
+    firewall-offline-cmd --zone=public --add-port=30000-32767/tcp && \
+    firewall-offline-cmd --zone=public --add-port=30000-32767/udp
+
+# Create a systemd unit to recursively make the root filesystem subtree
+# shared as required by OVN images
+RUN cat > /etc/systemd/system/microshift-make-rshared.service <<'EOF'
+[Unit]
+Description=Make root filesystem shared
+Before=microshift.service
+ConditionVirtualization=container
+[Service]
+Type=oneshot
+ExecStart=/usr/bin/mount --make-rshared /
+[Install]
+WantedBy=multi-user.target
+EOF
+RUN systemctl enable microshift-make-rshared.service
+```
+
+> **Important:**<br>
+> When building a container image, podman uses host subscription information and
+> repositories inside the container. With only `BaseOS` and `AppStream` system
+> repositories enabled by default, we need to enable the `rhocp` and `fast-datapath`
+> repositories for installing MicroShift and its dependencies. These repositories
```

Comment:
Does it need to be mentioned here that an additional subscription that includes "Red Hat Enterprise Linux for Real Time" might be needed, or is it out of scope  ?

Reply:
I think this is out of scope for getting started because we're not installing low-latency RPM and profile